src/Microsoft.Health.Core/RetryableInitializationOperation.cs

```diff
@@ -18,13 +19,20 @@ namespace Microsoft.Health.Core
     public class RetryableInitializationOperation : IDisposable
     {
         private readonly Func<Task> _operation;
+        private readonly bool _continueOnCapturedContext;
         private SemaphoreSlim _semaphore = new SemaphoreSlim(1, 1);
         private Task _task;
 
         public RetryableInitializationOperation(Func<Task> operation)
+            : this(operation, false)
+        {
+        }
+
+        public RetryableInitializationOperation(Func<Task> operation, bool continueOnCapturedContext)
```

Comment:
Because we don't make any assumptions about the task we're given, we should allow the user to specify the argument for `ConfigureAwait`. In fact, the lib Polly follows this pattern as well on its `ExecuteAsync` methods